Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -389,42 +311,69 @@ public void WaitForTasksCompletion(IEnumerable<string> taskIds)
     }
 
     /// <summary>
-    /// Get the result status of a list of taskId
+    /// Get the result status of a list of results
     /// </summary>
-    /// <param name="taskIds"></param>
+    /// <param name="resultIds">Collection of result ids</param>
     /// <param name="cancellationToken"></param>
     /// <returns>A ResultCollection sorted by Status Completed, Result in Error or missing</returns>
-    public ResultStatusCollection GetResultStatus(IEnumerable<string> taskIds, CancellationToken cancellationToken = default)
+    public ResultStatusCollection GetResultStatus(ICollection<string> resultIds, CancellationToken cancellationToken = default)
     {
       var idStatus = Retry.WhileException(5,
                                           200,
-                                          () =>
+                                          retry =>
                                           {
+                                            Logger.LogDebug("Try {try} for {funcName}",
+                                                            retry,
+                                                            nameof(ControlPlaneService.GetResultStatus));
                                             var resultStatusReply = ControlPlaneService.GetResultStatus(new GetResultStatusRequest()
                                             {
                                               ResultIds =
                                               {
-                                                taskIds,
+                                                resultIds,
                                               },
                                               SessionId = SessionId.Id,
                                             });
                                             return resultStatusReply.IdStatuses.Select(x => Tuple.Create(x.ResultId,
-                                                                                                       x.Status));
+                                                                                                         x.Status)).ToHashSet();
```

Comment:
You can put `.ToHashSet()` outside of the retry. It will be clearer what type `idStatus` is.